app/src/main/java/com/github/se/eduverse/repository/PdfRepository.kt

```diff
@@ -0,0 +1,272 @@
+          val words = line!!.split(" ")
+
+          // Draw the read line word by word into the pdf document
+          var currentLine = ""
+          for (word in words) {
+            // Check if the line with the next word fits in the pdf page width
+            val tempLine =
+                if (currentLine.isEmpty()) word
+                else "$currentLine $word" // Add the next word to the current line after a space,
+            // for the first line of the doc no space needed before
+            // the word
+            val textWidth = paint.measureText(tempLine)
+            if (textWidth <= pageInfo.pageWidth - 40) { // 20px padding on each side of the page
+              currentLine = tempLine
+            } else {
+              // Draw the current line and start a new line with the current word
+              canvas.drawText(currentLine, TEXT_PAGE_LEFT_PADDING, yPosition, paint)
+              yPosition += paint.descent() - paint.ascent()
+              currentLine = word
+            }
+
+            // If the current page is full, create a new page and add it to the pdf document
+            if (yPosition >
+                pageInfo.pageHeight - 80) { // 40px padding on top and bottom of the page
+              pdfDocument.finishPage(page)
+              yPosition = TEXT_PAGE_TOP_PADDING
+              val newPageInfo =
+                  PdfDocument.PageInfo.Builder(
+                          A4_PAGE_WIDTH, A4_PAGE_HEIGHT, pdfDocument.pages.size + 1)
+                      .create()
+              page = pdfDocument.startPage(newPageInfo)
+              canvas = page.canvas
+            }
+          }
+
+          // Draw the last word of the read line if it didn't fit in the page's previous line
+          if (currentLine.isNotEmpty()) {
+            canvas.drawText(currentLine, TEXT_PAGE_LEFT_PADDING, yPosition, paint)
+            yPosition += paint.descent() - paint.ascent()
+          }
+
+          // If the last word of the read line was written on the last line of the page, create a
+          // new page
+          if (yPosition > pageInfo.pageHeight - 40) {
+            pdfDocument.finishPage(page)
+            yPosition = TEXT_PAGE_TOP_PADDING
+            val newPageInfo =
+                PdfDocument.PageInfo.Builder(
+                        A4_PAGE_WIDTH, A4_PAGE_HEIGHT, pdfDocument.pages.size + 1)
+                    .create()
+            page = pdfDocument.startPage(newPageInfo)
+            canvas = page.canvas
+          }
+        }
+      }
+
+      // Finish the last page and write the pdf document to the destination file, display a toast
+      // if document creation is successful
+      pdfDocument.finishPage(page)
+
+      return pdfDocument
+    } catch (e: Exception) {
+      // If an exception occurs during the conversion process, log the error and throw the exception
+      Log.e("convertTextToPdf", "Text conversion failed", e)
+      throw e
+    }
+  }
+
```

Comment:
The PDF may not adhere to standard dimensions, which might be problematic for printing.